controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -398,32 +404,61 @@ public void createScope(ScopeInfo request, StreamObserver<CreateScopeStatus> res
     @Override
     public void listStreamsInScope(Controller.StreamsInScopeRequest request, StreamObserver<Controller.StreamsInScopeResponse> responseObserver) {
         String scopeName = request.getScope().getScope();
-        RequestTag requestTag = requestTracker.initializeAndTrackRequestTag(requestIdGenerator.get(), "listStream", scopeName);
+        RequestTag requestTag = requestTracker.initializeAndTrackRequestTag(requestIdGenerator.get(),
+                "listStream", scopeName);
         log.info(requestTag.getRequestId(), "listStream called for scope {}.", scopeName);
+
+        final AuthContext ctx;
+        if (this.grpcAuthHelper.isAuthEnabled()) {
+            ctx = AuthContext.current();
+        } else {
+            ctx = null;
+        }
+
         authenticateExecuteAndProcessResults(
-                () -> this.authHelper.checkAuthorization(AuthResourceRepresentation.ofScope(scopeName),
-                        AuthHandler.Permissions.READ),
-                delegationToken -> controllerService.listStreams(
-                        scopeName, request.getContinuationToken().getToken(), listStreamsInScopeLimit)
+                () -> {
+                        String result = this.grpcAuthHelper.checkAuthorization(
```

Comment:
why cant we simply call checkAuthorization as it internally takes the same context as it is computed here --> AuthContext.current()

Reply:
The gRPC Context isn't available at this stage of the operation and therefore the `Principal` and `AuthInterceptor` (erstwhile `PravegaInterceptor`) objects cannot be retrieved from the gRPC Context at this point. Since those values were null, we were defaulting to assuming the user doesn't have access to the stream. This was the root cause for the list streams filtering not returning any streams in scope with auth enabled. 

To avoid the issue, we now cache the relevant `Principal` and `AuthHandler` objects in the `AuthContext` object during the commencement of the operation when the gRPC Context is available. We then use the cached `AuthContext` for filtering later in the same operation.

Hope that clarifies.